query/workmanager.go

```diff
@@ -423,11 +437,32 @@ Loop:
 					batch.errChan <- ErrQueryTimeout
 					delete(currentBatches, batchNum)
 
+					// When deleting the particular batch
+					// number we need to make sure we delete
+					// all the ongoing queryJobs in queue.
+					// Otherwise these queries might never
+					// be removed (in case they fail because
+					// of a timeout for example).
+					//
+					// NOTE: We do NOT cancel ongoing
+					// queries with the workers because it's
+					// easier to let them timeout and just
+					// be dropped because their
+					// corresponding batch is already
+					// removed.
```

Comment:
Are we sure we want to leave dangling queries like this? I think I understand that it's not necessary for us to kill them, but I wonder if we have the capacity to deal with the results if we leave the queries. Are the outstanding queries guaranteed to timeout? If not, what happens to the results?

Reply:
Right but consider how much cognitive load is required to ascertain that. Instead we should just cancel that because now we are relying on behavior defined 100+ LOC away to mutate state in exactly the right way and check those conditions in exactly the right way.

Reply:
Oof. I don't want to scope creep this, but I want it on the record that relying on shared state for this kind of thing that isn't part of the API of the thing being operated on is not good practice.

Reply:
I think we can definitely remove the ongoing jobs as well (we just need to track the active work), is this what you referring to with the shared state ? I am not sure if I understood the concern in detail, may you elaborate ?

Reply:
I would disregard what I said. It is out of scope for this change. My concern is that we haven't layered components in neutrino properly and the code is very delicate and requires "whole program understanding" as opposed to being able to reason locally about what is going on. I'm fine with the choices made here as the parts I take issue with precede your changes.

Reply:
introduced another design so this comment is obsolete.